src/overlays/actors/ovl_En_Stone_heishi/z_en_stone_heishi.c

```diff
@@ -35,54 +39,496 @@ const ActorInit En_Stone_heishi_InitVars = {
+                    this->bottleDisplay = EN_STONE_BOTTLE_EMPTY;
+                    Actor_PlaySfxAtPos(&this->actor, NA_SE_VO_NP_DRINK);
+                    func_80123D50(play, GET_PLAYER(play), ITEM_BOTTLE, PLAYER_AP_BOTTLE);
+                }
+            } else {
+                this->drinkBottleState++;
+            }
+            break;
+
+        case EN_STONE_DRINK_BOTTLE_STAND_UP:
+            if (this->endFrame <= currentFrame) {
+                Audio_PlayFanfare(NA_BGM_GET_ITEM | 0x900);
+                this->bottleDisplay = EN_STONE_BOTTLE_NONE;
+                EnStoneheishi_ChangeAnim(this, EN_STONE_HEISHI_ANIM_STAND_UP);
+                this->drinkBottleState++;
+            }
+            break;
+
+        case EN_STONE_DRINK_BOTTLE_STANDING:
+            if (this->endFrame <= currentFrame) {
+                this->textIdIndex = 5;
+                func_80151938(play, sEnStoneHeishiTextIds[this->textIdIndex]);
+                player->actor.textId = sEnStoneHeishiTextIds[this->textIdIndex];
+                EnStoneheishi_ChangeAnim(this, EN_STONE_HEISHI_ANIM_STAND_HAND_ON_HIP);
+                this->action = EN_STONE_ACTION_1;
+                this->actionFunc = func_80BC9680;
+            }
+            break;
+    }
+}
+
+void EnStoneheishi_GiveItemPrize(EnStoneheishi* this, PlayState* play) {
+    func_801477B4(play);
+
+    if (INV_CONTENT(ITEM_MASK_STONE) == ITEM_MASK_STONE) {
+        Actor_PickUp(&this->actor, play, GI_RUPEE_BLUE, 300.0f, 300.0f);
+    } else {
+        Actor_PickUp(&this->actor, play, GI_MASK_STONE, 300.0f, 300.0f);
+    }
+
+    this->action = EN_STONE_ACTION_4;
+    this->actionFunc = func_80BC9D28;
+}
+
+void func_80BC9D28(EnStoneheishi* this, PlayState* play) {
+    SkelAnime_Update(&this->skelAnime);
+
+    if (Actor_HasParent(&this->actor, play)) {
+        this->actor.parent = NULL;
+        this->textIdIndex++;
+        this->actor.textId = sEnStoneHeishiTextIds[this->textIdIndex];
+        gSaveContext.save.weekEventReg[41] |= 0x40;
+        Actor_ProcessTalkRequest(&this->actor, &play->state);
+        func_800B8500(&this->actor, play, 400.0f, 400.0f, PLAYER_AP_MINUS1);
+        this->actionFunc = func_80BC9E50;
+    } else if (INV_CONTENT(ITEM_MASK_STONE) == ITEM_MASK_STONE) {
+        Actor_PickUp(&this->actor, play, GI_RUPEE_BLUE, 300.0f, 300.0f);
+    } else {
+        Actor_PickUp(&this->actor, play, GI_MASK_STONE, 300.0f, 300.0f);
+    }
+}
+
+void func_80BC9E50(EnStoneheishi* this, PlayState* play) {
+    SkelAnime_Update(&this->skelAnime);
+
+    if (Actor_ProcessTalkRequest(&this->actor, &play->state)) {
+        func_80151BB4(play, 0x35);
+        func_80151BB4(play, 0x12);
+        this->action = EN_STONE_ACTION_1;
+        this->actionFunc = func_80BC9680;
+    } else {
+        func_800B8500(&this->actor, play, 400.0f, 400.0f, PLAYER_AP_MINUS1);
+    }
+}
+
+void EnStoneheishi_Update(Actor* thisx, PlayState* play) {
+    EnStoneheishi* this = THIS;
+    s32 pad;
```

Comment:
```suggestion
    s32 pad;
    EnStoneheishi* this = THIS;
```
is more normal if it matches.